cmd/access-perms.go

```diff
@@ -29,7 +34,32 @@ func (b accessPerms) isValidAccessPERM() bool {
 }
 
 func (b accessPerms) isValidAccessFile() bool {
-	return filepath.Ext(string(b)) == ".json"
+	file, err := os.Open(string(b))
+	if err != nil {
+		fatalIf(errDummy().Trace(), "Unable to open access file.")
+		return false
+	}
+	defer file.Close()
+
+	var policy policy.BucketAccessPolicy
+	if json.NewDecoder(file).Decode(&policy) != nil {
+		fatalIf(errDummy().Trace(), "Unable to parse access file.")
+		return false
+	}
+
+	if policy.Version != "2012-10-17" {
+		fatalIf(errDummy().Trace(), "Invalid policy version. Only 2012-10-17 is supported.")
+		return false
+	}
+
+	for _, statement := range policy.Statements {
+		if statement.Effect != "Allow" && statement.Effect != "Deny" {
+			fatalIf(errDummy().Trace(), "Invalid policy effect. Only Allow and Deny are supported.")
+			return false
+		}
+	}
```

Comment:
I am not too much into client side validation; because the server already does that

Reply:
I think he is adding that to fail early when we do not know the file extension, its just a more stricter check.